### PR TITLE
build/configs/rtl8720e/scripts: Correct PSRAM Heap size and layout

### PR DESCRIPTION
**Pull request by ziyik**

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@ziyik Could you leave details in the commit description? What issue is, why it should be modified like this

**Comment by ziyik** [comment]
> @ziyik Could you leave details in the commit description? What issue is, why it should be modified like this

Previously PSRAM_HEAP_BASE is based on _ebss, it has been wrongly moved to under KM4_BD_RAM section.
And PSRAM layout section has been removed.
Thus, this commit is to add the PSRAM layout and correct the PSRAM_HEAP_BASE.
I will add the purpose in the commit message.

**Comment by sunghan-chang** [comment]
@kishore-sn @thapav Could you review heap configuration?

File: build/configs/rtl8720e/scripts/rlx8720e_img2_ns.ld [hunk 1]
```diff
@@ -83,7 +83,7 @@ MEMORY
 	/* IMG2 RAM */
 	KM4_BD_RAM (rwx) :                      ORIGIN = 0x20012020, LENGTH = 0x20040000 - 0x20012020	/* KM4 BD RAM: 184k */
 	KR4_BD_RAM (rwx) :                      ORIGIN = 0x20040060, LENGTH = 0x2007A000 - 0x20040060	/* KR4 BD RAM: 232k */
-	KM4_HEAP_EXT (rwx) :                    ORIGIN = 0x2007A000, LENGTH = 0x200B0000 - 0x2007A000	/* KM4 HEAP EXT: 280k(8720EA) */
+	KM4_HEAP_EXT (rwx) :                    ORIGIN = 0x2007A000, LENGTH = 0x200B0000 - 0x2007A000	/* KM4 HEAP EXT: 216k(8720EA) */
```
Comment (thapav) [inline build/configs/rtl8720e/scripts/rlx8720e_img2_ns.ld:86]: Do we not need to update the script for trustzone disabled scenario <rlx8720e_img2_is.ld> ?

File: os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c [hunk 2]
```diff
@@ -43,15 +43,15 @@ extern u32 GlobalDebugEnable;
 #ifdef CONFIG_PLATFORM_TIZENRT_OS
 extern unsigned int __StackStart;
 extern unsigned int __StackLimit;
-extern unsigned int _ebss;
+extern uint8_t __psram_heap_buffer_start__[];
```
Comment (thapav) [inline os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c:46]: Kindly check for build break in tc config >
/root/tizenrt/os/arch/arm/src/board/libboard.a(ameba_app_start.o): In function `os_heap_init':
/root/tizenrt/os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c:71: undefined reference to `__psram_heap_buffer_start__'
make[1]: *** [../build/output/bin/tinyara.axf] Error 1

File: build/configs/rtl8720e/scripts/rlx8720e_img2_ns.ld [hunk 3]
```diff
@@ -299,11 +299,24 @@ SECTIONS
 
 	.bdram_heap.start (NOLOAD):
 	{
-		_sbss = ABSOLUTE(.);
 		__bdram_heap_buffer_start__ = .;
-		_ebss = ABSOLUTE(.);
```
[the remark this answers is not in the document]
Comment (ziyik) [reply build/configs/rtl8720e/scripts/rlx8720e_img2_ns.ld:304]: I see, I will revert this change. Will keep the definition of _ebss and _ebss.

File: os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c (hunk 2 above)
Comment (ziyik) [reply os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c:46]: I did not encounter any build errors with TC config.
May I know how did you get this error?

**Comment by sunghan-chang** [comment]
@ziyik @kishore-sn Because heap configuration and idle stack allocation is not merged with Realtek way and TizenRT way, it looks something wrong.
in os_start, idle stack is assigned as below, 
```
	g_idletcb.cmn.adj_stack_size = CONFIG_IDLETHREAD_STACKSIZE;
	g_idletcb.cmn.stack_alloc_ptr = (void *)(g_idle_topstack - CONFIG_IDLETHREAD_STACKSIZE);
	g_idletcb.cmn.adj_stack_ptr = (void *)(g_idle_topstack - 4);
```
g_idle_topstack is assigned in amebalite_start.c
```
#define MSP_RAM_HP_NS		0x20003FFC
const uintptr_t g_idle_topstack = MSP_RAM_HP_NS;
```
It's not after _ebss
```
	KM4_BD_RAM (rwx) :                      ORIGIN = 0x20012020, LENGTH = 0x20040000 - 0x20012020	/* KM4 BD RAM: 184k */

	.bdram_heap.start (NOLOAD):
	{
		_sbss = ABSOLUTE(.);
		__bdram_heap_buffer_start__ = .;
		_ebss = ABSOLUTE(.);
	} > KM4_BD_RAM
```

And heap configuration on psram is not configured well either.
We should check this.

File: os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c (hunk 2 above)
Comment (thapav) [reply os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_app_start.c:46]: The code was not rebased. It seems to work fine. We can ignore this comment

**Review by thapav — APPROVED** [review]

File: build/configs/rtl8720e/hello/Make.defs [hunk 4]
```diff
@@ -54,11 +54,7 @@ include ${TOPDIR}/tools/Config.mk
 ARCH_FAMILY = $(patsubst "%",%,$(CONFIG_ARCH_FAMILY))
 include ${TOPDIR}/arch/$(CONFIG_ARCH)/src/$(ARCH_FAMILY)/Toolchain.defs
 
-ifeq ($(CONFIG_AMEBALITE_TRUSTZONE),y)
-LDSCRIPT = rlx8720e_img2_ns.ld
-else
-LDSCRIPT = rlx8720e_img2_is.ld
-endif
+LDSCRIPT = rlx8720e_img2.ld
```
Comment (thapav) [inline build/configs/rtl8720e/hello/Make.defs:57]: May I know why have we removed the other script and merged into one ?

**Review by thapav — COMMENTED** [review]
MAy I know why have we merged

File: build/configs/rtl8720e/hello/Make.defs (hunk 4 above)
Comment (ziyik) [reply build/configs/rtl8720e/hello/Make.defs:57]: TrustZone and Non-TrustZone used the same layout for TP1e.,
So we should maintain only one script and remove the other.

**Comment by sunghan-chang** [comment]
@ziyik Once @kishore-sn implements new variables for idle stack and psram heap, we would ask to apply them or would do by ourself.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**